### PR TITLE
Switch to GitHub Actions and build for aarch64

### LINKED ISSUE #28: Arm64 images on Docker Hub
It looks like only amd64 flavor of images is available on Docker hub. It would be nice to have an arm64 equivalent as well.

### PR DESCRIPTION
**Pull request by mmlb**
This PR enables building docker images for aarch64 machines. It does this by using `docker buildx`'s [support](https://github.com/docker/buildx/#building-multi-platform-images) for building multarch images.

I'm also transitioning the build from travis to github actions. Is it fine to do it in this PR or should  I split that off into a separate PR that this will block on?

This PR closes #28.

### CODE REVIEW

**Comment by mmlb** [comment]
Hi @peti, ping?

**Comment by peti** [comment]
I suppose it's fair to say that this project needs a new maintainer. I have been completely busy with other projects and cannot really contribute to this effort any more.

**Comment by mmlb** [comment]
I'd love to take up that work.

File: Dockerfile
Comment (domenkozar) [inline Dockerfile:17]: Why do we need filter calls now?

**Review by domenkozar — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (mmlb) [reply Dockerfile:17]: I couldn't remember the exact failure so tried to reproduce it again. When running docker build under qemu soft emulation nix just fails to install. I've updated the commit message with the info and a snipped of the failure here: [`609eab8` (#43)](https://github.com/NixOS/docker/pull/43/commits/609eab80faf4681e574323de35d1740eff52ba47)

**Comment by mmlb** [comment]
I've rebased on top of master to pick up `git`.

File: .github/workflows/build.yaml
Comment (domenkozar) [inline .github/workflows/build.yaml:20]: Is there no way to use a token?

Comment (mmlb) [reply .github/workflows/build.yaml:20]: As in a docker hub access token (https://docs.docker.com/docker-hub/access-tokens/)? I think if you setup the token and use that as the password it would all work out, at least thats my reading of the web page there.

**Comment by domenkozar** [comment]
I'm going to merge this unless someone objects in a few days.

We really need a docker image with Nix 2.4

**Comment by mmlb** [comment]
> I'm going to merge this unless someone objects in a few days.
> 
> We really need a docker image with Nix 2.4

Lets do it?

**Comment by domenkozar** [comment]
cc @garbas for https://github.com/NixOS/nix/pull/5453#issuecomment-970389396

**Comment by garbas** [comment]
@domenkozar tnx for the ping

@mmlb The pure Nix docker image was merged into `nixos/nix` repo (see https://github.com/NixOS/nix/pull/5453). There are also both [x86_64-linux](https://hydra.nixos.org/job/nix/master/dockerImage.x86_64-linux) and [aarch64-linux](https://hydra.nixos.org/job/nix/master/dockerImage.aarch64-linux) images being built by hydra. 

In #31 I asked @peti to share the permission to configure the github action to push to hub.docker.com for each commit (until then I'll just make the 2.4 release manually).

Once that is setup I think it would be better to archive this repo and update the readme.

Also if you have time any help is more than appreciated.

**Comment by mmlb** [comment]
> @domenkozar tnx for the ping
> 
> @mmlb The pure Nix docker image was merged into `nixos/nix` repo (see [NixOS/nix#5453](https://github.com/NixOS/nix/pull/5453)). There are also both [x86_64-linux](https://hydra.nixos.org/job/nix/master/dockerImage.x86_64-linux) and [aarch64-linux](https://hydra.nixos.org/job/nix/master/dockerImage.aarch64-linux) images being built by hydra. 

Nice.

> In #31 I asked @peti to share the permission to configure the github action to push to hub.docker.com for each commit (until then I'll just make the 2.4 release manually).
> 
> Once that is setup I think it would be better to archive this repo and update the readme.
 
I'm mostly concerned that there's no issue/pr to get pushes from nixos/nix to docker hub yet. Do you have any idea when that may happen? If it's up in the air I'd love to get this PR merged and then the docker images can then be switched over to coming from nixos/nix.

> Also if you have time any help is more than appreciated.

The nixos/nix setup is quite past my current nix level.

File: .github/workflows/build.yaml
Comment (vdovhanych) [reply .github/workflows/build.yaml:20]: Definitely use token for this. It's working with variables set like this.
```
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
```

File: Dockerfile
Comment (vdovhanych) [reply Dockerfile:17]: This was also an issue for me when trying to build an arm64 image with Github actions. If I built it on M1 MacBook the build finished fine and I ended up with an arm64 image that i could use.

**Comment by mmlb** [comment]
I think I'll close this PR in favor of https://github.com/NixOS/nix/pull/5453 and further work.

**Pull request closed without merging**
